panels/updates/cc-updates-panel.c

```diff
@@ -346,119 +428,80 @@ update_active_network (CcUpdatesPanel *self)
     }
-
-  /* Automatic Updates switch */
-  gtk_widget_set_sensitive (self->automatic_updates_container, connection != NULL);
-
-  ensure_setting_user (self, connection);
-
-  if (connection)
-    {
-      NMSettingUser *setting_user;
-      const gchar *value;
-
-      setting_user = NM_SETTING_USER (nm_connection_get_setting (connection, NM_TYPE_SETTING_USER));
-      value = nm_setting_user_get_data (setting_user, NM_SETTING_ALLOW_DOWNLOADS_WHEN_METERED);
-
-      g_signal_handlers_block_by_func (self->automatic_updates_switch,
-                                       on_automatic_updates_switch_changed_cb,
-                                       self);
-
-      gtk_switch_set_active (GTK_SWITCH (self->automatic_updates_switch), g_strcmp0 (value, "1") == 0);
-
-      g_signal_handlers_unblock_by_func (self->automatic_updates_switch,
-                                         on_automatic_updates_switch_changed_cb,
-                                         self);
-    }
 }
 
 static void
-load_tariff (CcUpdatesPanel *self)
+update_active_network (CcUpdatesPanel *self)
 {
-  g_autoptr(MwtTariffLoader) tariff_loader = NULL;
-  g_autoptr(GBytes) bytes = NULL;
-  g_autoptr(GError) error = NULL;
-  g_autofree gchar *tariff_path = NULL;
-  gchar *contents = NULL;
-  gsize size = 0;
+  NMAccessPoint *ap;
+  NMConnection *connection;
+  NMDevice *device;
+  const gchar *icon_name;
 
-  g_debug ("Loading tariff");
+  get_active_connection_and_device (self, &device, &connection, &ap);
 
-  tariff_path = g_build_filename (g_get_user_config_dir (), SYSTEM_TARIFF_FILENAME, NULL);
-  g_file_get_contents (tariff_path, &contents, &size, &error);
-  if (error)
-    {
-      if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
-        g_warning ("Error reading tariff file: %s", error->message);
-      return;
-    }
+  /* Setup the new device */
+  if (self->current_device != device)
+    cleanup_signals (self);
 
-  bytes = g_bytes_new_take (contents, size);
-  tariff_loader = mwt_tariff_loader_new ();
-  mwt_tariff_loader_load_from_bytes (tariff_loader, bytes, &error);
-  if (error)
+  if (g_set_object (&self->current_device, device) && device)
     {
-      g_warning ("Error loading tariff: %s", error->message);
-      return;
+      self->changed_id = g_signal_connect_swapped (device,
+                                                   "state-changed",
+                                                   G_CALLBACK (on_network_changed_cb),
+                                                   self);
     }
 
-  g_signal_handlers_block_by_func (self->tariff_editor, on_tariff_changed_cb, self);
-  g_signal_handlers_block_by_func (self->scheduled_updates_switch, on_scheduled_updates_switch_changed_cb, self);
+  /* Icon */
+  icon_name = get_network_status_icon_name (self, device, ap);
+  gtk_image_set_from_icon_name (GTK_IMAGE (self->network_status_icon), icon_name, 4);
```

Comment:
s/`4`/`GTK_ICON_SIZE_BUTTON`/? (I think that’s the right enum value; please check.)